src/VisualStudio/Core/Def/Implementation/IntellisenseControls/IntellisenseTextBox.cs

```diff
@@ -0,0 +1,528 @@
+            // depending on the current language. Override the font information
+            // to have uniform look and feel in the parallel watch window
+            IClassificationFormatMapService classificationFormatMapService = componentModel.GetService<IClassificationFormatMapService>();
+            IClassificationFormatMap classificationFormatMap = classificationFormatMapService.GetClassificationFormatMap(appearanceCategory);
+            Typeface typeface = new Typeface(container.FontFamily, container.FontStyle, container.FontWeight, container.FontStretch);
+            classificationFormatMap.DefaultTextProperties = classificationFormatMap.DefaultTextProperties.SetTypeface(typeface);
+            classificationFormatMap.DefaultTextProperties = classificationFormatMap.DefaultTextProperties.SetFontRenderingEmSize(container.FontSize);
+
+            // Install this object in the text view command filter chain
+            IEditorOperationsFactoryService editorOperationsFactoryService = componentModel.GetService<IEditorOperationsFactoryService>();
+            if (editorOperationsFactoryService != null)
+            {
+                this._editorOperations = editorOperationsFactoryService.GetEditorOperations(viewModel.WpfTextView);
+            }
+
+            ErrorHandler.ThrowOnFailure(viewModel.VsTextView.AddCommandFilter(this, out this._nextCommandTarget));
+
+            // Get the host control to render the view
+            IVsEditorAdaptersFactoryService editorAdapterFactory = componentModel.GetService<IVsEditorAdaptersFactoryService>();
+            this._textViewHost = editorAdapterFactory.GetWpfTextViewHost(viewModel.VsTextView);
+
+            BindingOperations.SetBinding(_textViewHost.HostControl, AutomationProperties.NameProperty, new Binding { Source = container, Path = new PropertyPath(AutomationProperties.NameProperty) });
+            BindingOperations.SetBinding(_textViewHost.HostControl, AutomationProperties.LabeledByProperty, new Binding { Source = container, Path = new PropertyPath(AutomationProperties.LabeledByProperty) });
+
+            // For non-blurry text
+            TextOptions.SetTextFormattingMode(this._textViewHost.HostControl, TextFormattingMode.Display);
+            this._textViewHost.HostControl.Loaded += this.HostControl_Loaded;
+
+            this.AddLogicalChild(this._textViewHost.HostControl);
+            this.AddVisualChild(this._textViewHost.HostControl);
+        }
+
+        /// <summary>
+        /// Event handler for editor control load
+        /// </summary>
+        /// <param name="sender">Editor control</param>
+        /// <param name="e">Event args</param>
+        private void HostControl_Loaded(object sender, RoutedEventArgs e)
+        {
+            if (this._textViewHost != null)
+            {
+                // Set the focus
+                this._textViewHost.TextView.VisualElement.Focus();
+
+                if (this._textViewHost != null)
```

Comment:
Why the nested null checking? (I'm also not sure how it can ever be null given the thing is set to not-null in InitializeEditorControl and repeatedly referenced!)